src/aggregation/segment_agg_result.rs

```diff
@@ -177,10 +186,20 @@ impl SegmentBucketResultCollector {
             SegmentBucketResultCollector::Range(range) => {
                 range.collect_block(doc, bucket_with_accessor, force_flush);
             }
+            SegmentBucketResultCollector::Histogram(histogram) => {
+                histogram.collect_block(doc, bucket_with_accessor, force_flush)
+            }
         }
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct SegmentHistogramBucketEntry {
+    pub key: f64,
```

Comment:
Why do we need the key in here?

Reply:
we could compute it at the end before filtering the collector for 0 buckets, because we implicitly know which bucket index has which key, but I think I'd not do this optimization now, since it will limit how we can change the algorithm